egs/librispeech/ASR/conformer_ctc/gigaspeech_datamodule.py

```diff
@@ -0,0 +1,403 @@
+                cuts_train,
+                max_duration=self.args.giga_max_duration,
+                shuffle=self.args.giga_shuffle,
+            )
+        logging.info("About to create train dataloader")
+        #train_dl = DataLoader(
+        #    train,
+        #    sampler=train_sampler,
+        #    batch_size=None,
+        #    num_workers=16,
+        #    persistent_workers=True,
+        #)
+        train_dl = LhotseDataLoader(
+            train,
+            sampler=train_sampler,
+            num_workers=3,
+            prefetch_factor=5,
+        )
+        return train_dl
+
+    def inexhaustible_train_dataloaders(self):
+        return self
+
+    def valid_dataloaders(self) -> DataLoader:
+        self.validate_args()
+        logging.info("About to get dev cuts")
+        cuts_valid = self.valid_cuts()
+
+        transforms = [ ]
+        if self.args.giga_concatenate_cuts:
+            transforms = [ CutConcatenate(
+                                 duration_factor=self.args.giga_duration_factor,
+                                 gap=self.args.giga_gap)
+                          ] + transforms
+
+
+        logging.info("About to create dev dataset")
+        if self.args.giga_on_the_fly_feats:
+            validate = K2SpeechRecognitionDataset(
+                cuts_valid,
+                cut_transforms=transforms,
+                input_strategy=OnTheFlyFeatures(Fbank(FbankConfig(num_mel_bins=80)), num_workers=8),
+                return_cuts=True,
+                check_inputs=self.args.giga_check_cuts,
+            )
+        else:
+            validate = K2SpeechRecognitionDataset(
+                # cuts_valid,
+                cut_transforms=transforms,
+                return_cuts=True,
+                # check_inputs=self.args.giga_check_cuts,
+            )
+        valid_sampler = SingleCutSampler(
+            cuts_valid,
+            max_duration=self.args.giga_max_duration,
+            shuffle=False,
+        )
+        logging.info("About to create dev dataloader")
+        #valid_dl = DataLoader(
+        #    validate,
+        #    sampler=valid_sampler,
+        #    batch_size=None,
+        #    num_workers=8,
+        #    persistent_workers=True,
+        #)
+        valid_dl = LhotseDataLoader(
+            validate,
+            sampler=valid_sampler,
+            num_workers=2,
+        )
+        return valid_dl
+
+    def test_dataloaders(self) -> Union[DataLoader, List[DataLoader]]:
+        self.validate_args()
+        cuts = self.test_cuts()
+        is_list = isinstance(cuts, list)
+        test_loaders = []
+        if not is_list:
+            cuts = [cuts]
+
+        for cuts_test in cuts:
+            logging.debug("About to create test dataset")
+            test = K2SpeechRecognitionDataset(
+                cuts_test,
+                input_strategy=(
+                    OnTheFlyFeatures(Fbank(FbankConfig(num_mel_bins=80)), num_workers=8)
```

Comment:
Note: replacing this with `KaldifeatFbank` in latest Lhotse may speed up on-the-fly feature extraction by 20~30% (on CPU).